poc_iot_injector/src/cli/generate.rs

```diff
@@ -33,24 +36,41 @@ impl Cmd {
         let poc_oracle_key = settings.keypair()?;
         let shared_key = Arc::new(poc_oracle_key);
 
+        let success_counter = Arc::new(Mutex::new(0));
+        let failure_counter = Arc::new(Mutex::new(0));
+
         store
             .source_unordered(LOADER_WORKERS, stream::iter(file_list).map(Ok).boxed())
-            .for_each_concurrent(STORE_WORKERS, |msg| {
+            .try_for_each_concurrent(STORE_WORKERS, |msg| {
                 let shared_key_clone = shared_key.clone();
+                let success_counter_ref = Arc::clone(&success_counter);
+                let failure_counter_ref = Arc::clone(&failure_counter);
                 async move {
-                    match msg {
-                        Ok(m) => process_msg(m, shared_key_clone, before_ts).await,
-                        Err(e) => tracing::error!("unable to process msg due to: {:?}", e),
+                    if process_msg(msg, shared_key_clone, before_ts)
+                        .await
+                        .is_some()
+                    {
+                        *success_counter_ref.lock().unwrap() += 1;
+                    } else {
+                        *failure_counter_ref.lock().unwrap() += 1;
                     }
+                    Ok(())
                 }
             })
-            .await;
+            .await?;
+
+        tracing::debug!("success_counter: {:?}", success_counter.lock().unwrap());
+        tracing::debug!("failure_counter: {:?}", failure_counter.lock().unwrap());
 
         Ok(())
     }
 }
 
-async fn process_msg(msg: prost::bytes::BytesMut, shared_key_clone: Arc<Keypair>, before_ts: i64) {
+async fn process_msg(
```

Comment:
if we are moving to a `try_` version of the stream processor, would it be more idiomatic for this function to return a result rather than an option? the signature for these closures doesn't look like they expect the function to yield an option, so perhaps this is a carryover from an earlier version where we were iterating/streaming these synchronously?

Reply:
Yeah, simplified in [6e24642](https://github.com/helium/oracles/pull/167/commits/6e2464299930875f0e1d95b0e5d7f23449815140). Don't really like the `Result<Option<Thing>>` pattern, there's a specific `TxnConstruction` error enum now, and the result type is simplified to reflect the same.